test_othermlpackages/package_test_common.py

```diff
@@ -29,42 +33,88 @@ def fetch_adult():
+
+    unmitigated = copy.deepcopy(estimator)
+    unmitigated.fit(X_train, Y_train)
 
     expgrad = ExponentiatedGradient(
         estimator,
         constraints=moment)
-    expgrad.fit(X, Y, sensitive_features=A)
+    expgrad.fit(X_train, Y_train, sensitive_features=A_train)
 
     assert expgrad.n_oracle_calls_ > 1
     assert len(expgrad.predictors_) > 1
 
+    verification_moment.load_data(X_test, Y_test, sensitive_features=A_test)
+    gamma_unmitigated = verification_moment.gamma(lambda x: unmitigated.predict(x))
+    gamma_mitigated = verification_moment.gamma(lambda x: expgrad.predict(x))
+
+    for idx in gamma_mitigated.index:
+        assert abs(gamma_mitigated[idx]) <= abs(gamma_unmitigated[idx]), "Checking {0}".format(idx)
+
 
 def run_gridsearch_classification(estimator, moment):
     """Run classification test with GridSearch."""
-    X, Y, A = fetch_adult()
+    X_train, Y_train, A_train, X_test, Y_test, A_test = fetch_adult()
+    verification_moment = copy.deepcopy(moment)
 
-    num_predictors = 5
+    unmitigated = copy.deepcopy(estimator)
+    unmitigated.fit(X_train, Y_train)
+
+    num_predictors = 11
     gs = GridSearch(
         estimator,
         constraints=moment,
         grid_size=num_predictors)
-    gs.fit(X, Y, sensitive_features=A)
+    gs.fit(X_train, Y_train, sensitive_features=A_train)
 
     assert len(gs.predictors_) == num_predictors
 
+    verification_moment.load_data(X_test, Y_test, sensitive_features=A_test)
+    gamma_unmitigated = verification_moment.gamma(lambda x: unmitigated.predict(x))
+    gamma_mitigated = verification_moment.gamma(lambda x: gs.predict(x))
+
+    for idx in gamma_mitigated.index:
+        assert abs(gamma_mitigated[idx]) <= abs(gamma_unmitigated[idx]), "Checking {0}".format(idx)
+
 
 def run_thresholdoptimizer_classification(estimator):
     """Run classification test with ThresholdOptimizer."""
-    X, Y, A = fetch_adult()
+    X_train, Y_train, A_train, X_test, Y_test, A_test = fetch_adult()
+
+    unmitigated = copy.deepcopy(estimator)
+    unmitigated.fit(X_train, Y_train)
+    unmitigated_predictions = unmitigated.predict(X_test)
 
     to = ThresholdOptimizer(estimator=estimator, prefit=False)
-    to.fit(X, Y, sensitive_features=A)
+    to.fit(X_train, Y_train, sensitive_features=A_train)
+
+    mitigated_predictions = to.predict(X_test, sensitive_features=A_test)
+
+    dp_diff_unmitigated = demographic_parity_difference(Y_test,
+                                                        unmitigated_predictions,
+                                                        sensitive_features=A_test)
 
-    results = to.predict(X, sensitive_features=A)
-    assert results is not None
+    dp_diff_mitigated = demographic_parity_difference(Y_test,
+                                                      mitigated_predictions,
+                                                      sensitive_features=A_test)
+    assert dp_diff_mitigated <= dp_diff_unmitigated
```

Comment:
You can also assert that the disparity on the training data is 0 (or rather super close)